src/Microsoft.Sbom.Api/Executors/ManifestFolderFilterer.cs

```diff
@@ -16,15 +15,17 @@ namespace Microsoft.Sbom.Api.Executors
     /// </summary>
     public class ManifestFolderFilterer
     {
-        private readonly IFilter manifestFolderFilter;
+        private readonly IFilter<ManifestFolderFilter> manifestFolderFilter;
         private readonly ILogger log;
 
         public ManifestFolderFilterer(
-                                      [Named(nameof(ManifestFolderFilter))] IFilter manifestFolderFilter,
+                                      IFilter<ManifestFolderFilter> manifestFolderFilter,
                                       ILogger log)
         {
-            this.manifestFolderFilter = manifestFolderFilter ?? throw new ArgumentNullException(nameof(manifestFolderFilter));
-            this.log = log ?? throw new ArgumentNullException(nameof(log));
+            ArgumentNullException.ThrowIfNull(manifestFolderFilter);
```

Comment:
Is it now the recommended way to do null argument checks? There are so much different ways I'm lost which one is considered best-practice now :)